op-challenger/scripts/visualize.sh

```diff
@@ -2,30 +2,19 @@
 
 set -euo pipefail
 
-if [ $# -eq 0 ]
-  then
-    echo "Missing Fault Dispute Game address argument"
-fi
-
-echo ""
-echo "Visualize the fault dispute game at https://dispute.clab.by/game?addr=$1"
-echo ""
-
-DISPUTE_GAME_PROXY=$(jq .DisputeGameFactoryProxy .devnet/addresses.json)
-DISPUTE_GAME_PROXY=$(echo $DISPUTE_GAME_PROXY | tr -d '"')
-
-echo "----------------------------------------------------------------"
-echo " Dispute Game Factory at $DISPUTE_GAME_PROXY"
-echo "----------------------------------------------------------------"
-
-FAULT_GAME_ADDRESS=$1
-
-DIR=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)
-DIR=$(echo ${DIR%/*/*})
-cd $DIR/packages/contracts-bedrock
-
-forge script scripts/FaultDisputeGameViz.s.sol \
-  --sig "remote(address)" $FAULT_GAME_ADDRESS \
-  --fork-url http://localhost:8545
-
-mv dispute_game.svg "$dir"
+RPC=${1:?Must specify RPC address}
+GAME_ADDR=${2:?Must specify fault dispute game address}
+
+COUNT=$(cast call --rpc-url "${RPC}" "${GAME_ADDR}" 'claimDataLen() returns(uint256)')
+echo "Claim count: ${COUNT}"
+((COUNT=COUNT-1))
+for i in $(seq 0 "${COUNT}")
+do
+  CLAIM=$(cast call --rpc-url "${RPC}" "${GAME_ADDR}" 'claimData(uint256) returns(uint32 parentIndex, bool countered, bytes32 claim, uint128 position, uint128 clock)' "${i}")
+  SAVEIFS=$IFS   # Save current IFS (Internal Field Separator)
+  IFS=$'\n'      # Change IFS to newline char
+  CLAIM=($CLAIM) # split the string into an array by the same name
+  IFS=$SAVEIFS   # Restore original IFS
+
+  echo "${i}  Parent: ${CLAIM[0]} Countered: ${CLAIM[1]} Claim: ${CLAIM[2]} Position: ${CLAIM[3]} Clock ${CLAIM[4]}"
+done
```

Comment:
This seems like the wrong script to make these changes?

The visualize script was useful specifically for generating the game visualization. These changes seem to just print the game information. I don't think it makes sense to delete the visualize script logic and replace it with non-"visualizing" logic.

Reply:
I replaced the old script because it doesn't work at all with cannon games:
```
  [29849] FaultDisputeGameViz::remote(0x78A3f18f29fc7f1B92924C0Fd1ec4b628BE6aCca)
    ├─ [0] VM::load(0x78A3f18f29fc7f1B92924C0Fd1ec4b628BE6aCca, 0x0000000000000000000000000000000000000000000000000000000000000001) [staticcall]
    │   └─ ← 0x883ab54109c90c676cdac086acec1641312ad5135fc05d999e2197b3d55ee2f5
    └─ ← "Memory allocation overflow"
```

I've brought it back with a couple of tweaks and made this new script `list_claims.sh` but I worry that the visualize script will just cause confusion given it doesn't actually work.